File: KOCMOC.Server/Models/Entities/Producer.cs
using KOCMOC.Server.Models.Interfaces;
using KOCMOC.Server.Models.Enums;

namespace KOCMOC.Server.Models.Entities
{
  public class Producer : IAuditableEntity
  {
    public int Id { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? ModifiedDate { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public string? ImageURL { get; set; }
    public ProducerRole ProducerRole { get; set; }
    public ProducerStatus ProducerStatus { get; set; }

    public Producer(string name)
    {
      Name = name;
    }
  }
}

File: KOCMOC.Client/src/app/models/entities/producer.ts
import { ProducerRole } from '../enums/producerRole';
import { ProducerStatus } from '../enums/producerStatus';

export class Producer {
  id: number;
  createdDate: Date;
  modifiedDate: Date | null;
  name: string;
  description: string | null;
  imageURL: string | null;
  producerRole: ProducerRole;
  producerStatus: ProducerStatus;
}
